### PR TITLE
STREAMS-1237 doc(install): add documentation for Kafka security settings

### PR DESCRIPTION
**Pull request by jfmarquet**
Thank you for your contribution to the Axway Streams documentation.

## Describe the changes

Add documentation for Kafka security settings: SASL authentication and TLS encryption.

## Checklist for contributors

Before submitting this PR, please make sure:

* [x] You have read the [contribution guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/)
* [x] You have signed the [Axway CLA](https://cla.axway.com/)
* [x] You have verified the technical accuracy of your change
* [x] You have verified that your change does not expose any sensitive information (passwords, keys, etc.)
* [x] You have followed the [Markdown guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/writing_markdown/)  (unless this is is a Netlify CMS contribution)

_Put an x in the boxes that apply. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your change._

## Checklist for approvers

_This section is to be filled out by project maintainers only._

Before approving this PR, please make sure it:

* [x] Does not have conflicts with the base branch
* [x] Is believed to be accurate (technical accuracy to be checked with an SME for PRs originating from outside R&D)
* [x] Follows [Axway style](https://techweb.axway.com/confluence/display/RIE/Style+guide)
* [x] Follows [Markdown guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/writing_markdown/)
* [x] Follows [best practices](https://axway-open-docs.netlify.com/docs/contribution_guidelines/bestpracticedevdoc/)
* [x] Does not contain typos, grammatical errors, etc.
* [x] Does not expose any sensitive information
* [x] Passes the Markdown linter rules (automated via CI check)
* [x] Does not contain broken links

### CODE REVIEW

File: content/en/docs/install/_index.md [hunk 1]
```diff
@@ -139,6 +139,75 @@ To disable database encryption **and** TLS (not recommended for production use),
...
+export NAMESPACE="my-namespace"
+export KAFKA_CLIENT_PASSWORD="my-kakfa-client-password"
+export KAFKA_INTERBROKER_PASSWORD="my-kakfa-interbroker-password"
+
+kubectl -n ${NAMESPACE} create secret generic streams-kafka-credentials-secret --from-literal="client-passwords=${KAFKA_INTERBROKER_PASSWORD},${KAFKA_CLIENT_PASSWORD}" --from-literal="inter-broker-password=${KAFKA_INTERBROKER_PASSWORD}" --from-literal="client-password=${KAFKA_CLIENT_PASSWORD}"
+```
+
+#### TLS encryption
+
+In order to configure TLS encryption, you need to have a valid truststore and one certificate per brokers. They must be all integrated into Java Key Stores (JKS) files.
```
Comment (allandenis38) [inline content/en/docs/install/_index.md:161]: per broker (remove "s")

File: content/en/docs/install/_index.md [hunk 2]
```diff
@@ -139,6 +139,75 @@ To disable database encryption **and** TLS (not recommended for production use),
 * `mariadb.tls.enabled` and `mariadb.encryption.enabled` to `false`
 * `mariadb.master.extraEnvVarsSecret` and `mariadb.slave.extraEnvVarsSecret` to `null`
 
+### Kafka security settings
+
+By default, Kafka is configured with SASL authentication and TLS encryption for clients and brokers.
```
Comment (allandenis38) [inline content/en/docs/install/_index.md:144]: Is there any link available to the official Kafka doc for SASL authentication and TLS encryption that we could add for customers looking for further details?

File: content/en/docs/install/_index.md (hunk 1 above)
Comment (allandenis38) [inline content/en/docs/install/_index.md:161]: They must be all integrated => They must all be integrated

File: content/en/docs/install/_index.md [hunk 3]
```diff
@@ -139,6 +139,75 @@ To disable database encryption **and** TLS (not recommended for production use),
...
+export KAFKA_INTERBROKER_PASSWORD="my-kakfa-interbroker-password"
+
+kubectl -n ${NAMESPACE} create secret generic streams-kafka-credentials-secret --from-literal="client-passwords=${KAFKA_INTERBROKER_PASSWORD},${KAFKA_CLIENT_PASSWORD}" --from-literal="inter-broker-password=${KAFKA_INTERBROKER_PASSWORD}" --from-literal="client-password=${KAFKA_CLIENT_PASSWORD}"
+```
+
+#### TLS encryption
+
+In order to configure TLS encryption, you need to have a valid truststore and one certificate per brokers. They must be all integrated into Java Key Stores (JKS) files.
+Be careful as each broker needs its own keystore and a different CN name matching the Kafka pod hostname as described in [bitnami documentation](https://github.com/bitnami/charts/tree/master/bitnami/kafka#enable-security-for-kafka-and-zookeeper).
+We provide you with a script to help with truststore and keystore generation (this is actually a patch of the script in bitnami documentation in order to deal properly with kubernetes deployment).
```
Comment (allandenis38) [inline content/en/docs/install/_index.md:163]: (this is actually a patch of the script in bitnami documentation in order to deal properly with kubernetes deployment) => (patched version of bitnami's script that properly handles Kubernetes deployment).

File: content/en/docs/install/_index.md [hunk 4]
```diff
@@ -139,6 +139,75 @@ To disable database encryption **and** TLS (not recommended for production use),
...
+You can also use your own truststore/privatekey:
+
+```sh
+cd tools
+./kafka-generate-ssl.sh
+```
+
+Then, create a secret which contains all the previously generated files
```
Comment (allandenis38) [inline content/en/docs/install/_index.md:171]: Missing ":"

File: content/en/docs/install/_index.md [hunk 5]
```diff
@@ -139,6 +139,75 @@ To disable database encryption **and** TLS (not recommended for production use),
...
+
+```sh
+export NAMESPACE="my-namespace"
+export KAFKA_SECRET_PASSWORD="my-kakfa-secret-password"
+kubectl -n ${NAMESPACE} create secret generic streams-kafka-client-jks-secret --from-file="./truststore/kafka.truststore.jks" --from-file=./keystore/kafka-0.keystore.jks --from-file=./keystore/kafka-1.keystore.jks --from-file=./keystore/kafka-2.keystore.jks --from-literal="jks-password=${KAFKA_SECRET_PASSWORD}"
+```
+
+#### Disable security
```
Comment (allandenis38) [inline content/en/docs/install/_index.md:179]: Disable Kafka security features

File: content/en/docs/install/_index.md [hunk 6]
```diff
@@ -139,6 +139,75 @@ To disable database encryption **and** TLS (not recommended for production use),
...
+
+To disable security you need to find & replace in `values.yaml` and `values-ha.yaml` all occurences of the following:
```
Comment (allandenis38) [inline content/en/docs/install/_index.md:181]: To disable Kafka security features

File: content/en/docs/install/_index.md [hunk 7]
```diff
@@ -139,6 +139,75 @@ To disable database encryption **and** TLS (not recommended for production use),
...
+    saslInterBrokerMechanism: plain
+```
+
+{{< alert title="Note" >}}
+Do not use for production.
+{{< /alert >}}
+
 ### Ingress TLS settings
 
 SSL/TLS is enabled by default on our embedded Ingress controller. If you don't provide any certificate, SSL will be enabled thanks to a nginx embedded fake SSL certificate.
```
Comment (allandenis38) [inline content/en/docs/install/_index.md:213]: SSL/TLS is enabled by default on **the** embedded Ingress controller

Comment (allandenis38) [inline content/en/docs/install/_index.md:213]: **fake** SSL certificate
Is it really fake or self-signed?

**Review by allandenis38 — CHANGES_REQUESTED** [review]

File: content/en/docs/install/_index.md (hunk 3 above)
Comment (jfmarquet) [reply content/en/docs/install/_index.md:163]: this is not a patch anymore as there are a lot of modification but I will keep your wording like this: based on bitnami's script that properly handles Kubernetes deployment

File: content/en/docs/install/_index.md (hunk 7 above)
Comment (jfmarquet) [reply content/en/docs/install/_index.md:213]: not my business but ok I'll make the change ;)

Comment (jfmarquet) [reply content/en/docs/install/_index.md:213]: It is really a fake certificate which is self signed but this is it does shows up as "fake certificate" when you have a look at it.

**Comment by jfmarquet** [comment]
@allandenis38 all your comments have been taken into account, you can review them once again.

**Review by allandenis38 — APPROVED** [review]

**Pull request merged**